pkg/server/sqlflowserver_test.go

```diff
@@ -140,6 +140,9 @@ func TestSQL(t *testing.T) {
 }
 
 func TestGoroutineLeaky(t *testing.T) {
+	// NOTE(typhoonzero): the server should consume all messages in the pipe before returns error
```

Comment:
Why does the server need to consume all messages? `sf.Pipe` supports closing on both ends.

The server side goroutine writes a message to `func (w *PipeWriter) Write(item interface{}) error`. If an error has returned, the goroutine should exit.

Please refer to this [design doc](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/close_producer_from_consumer.md) for details.

Reply:
Add `defer rd.Close()` solved this, thanks!

Reply:
Not a problem.